pkg/ring/kv/consul/client.go

```diff
@@ -207,13 +208,15 @@ func (c *Client) WatchKey(ctx context.Context, key string, f func(interface{}) b
 			continue
 		}
 
-		out, err := c.codec.Decode(kvp.Value)
-		if err != nil {
-			level.Error(util.Logger).Log("msg", "error decoding key", "key", key, "err", err)
-			continue
-		}
-		if !f(out) {
-			return
+		if kvp != nil {
```

Comment:
Could be outdented.
```
if kp == nil {
  log.Info("value is nil")
  continue
}

//logic.
```

Notice that I also added a info log line.

Reply:
Done